server/src/main/java/org/candlepin/resource/OwnerProductResource.java

```diff
@@ -458,19 +463,23 @@ public void deleteProduct(
     @Produces(MediaType.APPLICATION_JSON)
     @Consumes(MediaType.WILDCARD)
     @Transactional
-    public JobDetail refreshPoolsForProduct(
+    public AsyncJobStatusDTO refreshPoolsForProduct(
         @PathParam("owner_key") String ownerKey,
         @PathParam("product_id") String productId,
-        @QueryParam("lazy_regen") @DefaultValue("true") Boolean lazyRegen) {
+        @QueryParam("lazy_regen") @DefaultValue("true") boolean lazyRegen) throws JobException {
 
         if (config.getBoolean(ConfigProperties.STANDALONE)) {
             log.warn("Ignoring refresh pools request due to standalone config.");
             return null;
         }
 
-        Owner owner = this.getOwnerByKey(ownerKey);
-        Product product = this.fetchProduct(owner, productId);
+        final Owner owner = this.getOwnerByKey(ownerKey);
+        final Product product = this.fetchProduct(owner, productId);
 
-        return RefreshPoolsForProductJob.forProduct(product, lazyRegen);
+        return this.translator.translate(
```

Comment:
It'll probably be cleanest to declare each step independently. Our style requirements are silly when it comes to multiline indentation, and there's really no reason to cram everything into one line here.

Reply:
Ok. Done.